htmresearch/support/junit_testing.py

```diff
@@ -53,33 +50,61 @@ def instantiateModel(args):
   return model
 
 
-def _trainModel(args, model, trainingData, labelRefs):
+def trainModel(model, trainingData, labelRefs, verbosity=0):
   """
   Train the given model on trainingData. Return the trained model instance.
   """
   print
   print "======================Training model on sample text==================="
-  if args.verbosity > 0:
-    printTemplate = "{0:<75}|{1:<20}|{2:<5}"
-    print printTemplate.format("Document", "Label", "ID")
+  if verbosity > 0:
+    printTemplate = PrettyTable(["ID", "Document", "Label"])
+    printTemplate.align = "l"
+    printTemplate.header_style = "upper"
   for (document, labels, docId) in trainingData:
-    if args.verbosity > 0:
-      print printTemplate.format(
-        wrapper.fill(document), labelRefs[labels[0]], docId)
+    if verbosity > 0:
+      printTemplate.add_row([docId, document, labelRefs[labels[0]]])
     model.trainDocument(document, labels, docId)
+  if verbosity > 0:
+    # printTemplate.set_style(align="l")
+    print printTemplate
 
   return model
 
 
-def _testModel(args, model, testData, labelRefs):
+def setupExperiment(args):
+  """
+  Create model according to args, train on training data, save model,
+  restore model.
+
+  @return newModel (ClassificationModel) The restored NLP model.
+  @return dataSet (list) Each item is a list representing a data sample, with
+      the text string, list of label indices, and the sample ID.
+  """
+  dataSet, labelRefs, _, _ = readDataAndReshuffle(args)
+  args.numLabels = len(labelRefs)
+
+  # Create a model, train it, save it, reload it
+  model = instantiateModel(args)
+  model = trainModel(model, dataSet, labelRefs, args.verbosity)
+  model.save(args.modelDir)
+  newModel = ClassificationModel.load(args.modelDir)
+
+  return newModel, dataSet
+
+
+def testModel(model, testData, categorySize, verbosity=0):
   """
-  Test the given model on testData, print out and return results metrics.
+  Test the given model on testData, print out and return results metrics. The
+  categorySize specifies the number of documents per category.
 
   For each data sample in testData the model infers the similarity to each other
-  sample; distances are number of bits apart. We then caclulate two types of
+  sample; distances are number of bits apart. We then calculate two types of
   results metrics: (i) "degrees of separation" and (ii) "overall ranks".
 
-    i. For the test document we want the distances for each "degree of
+    i. Doc #403 is separated from doc #s 402 and 404 by one degree, from doc #s
+    401 and 405 by two degrees, and so on. The fewer degrees of separation, the
+    more similarity we expect.
+    For the test document we want the distances for each "degree of
```

Comment:
For initial runs of these tests the ranks will suffice, but the degrees of separation metric gives us much more info. See [this comment](https://github.com/numenta/nupic.research/pull/455#issuecomment-184062960)... with the separation metric we can see that changing one word moves the samples ~60 bits apart from one another, and changing 5 words moves them 252 bits apart.

B/c this may be useful I'm hesitant to delete it yet, so I've moved it out to a separate method in order to keep the `testModel()` simple. If you feel we'll never use it, I can scrap it.